src/gendersprachekorrigieren.ts

```diff
@@ -103,92 +138,103 @@ export class BeGone {
-        if (BeGoneSettingsHelper.isWhitelist(this.settings) && RegExp(BeGoneSettingsHelper.whiteliststring(this.settings)).test(document.URL)) {
+        if (BeGoneSettingsHelper.isWhitelist(this.settings) && BeGoneSettingsHelper.whitelistRegexp(this.settings).test(document.URL)) {
             // White listed
             return true;
         }
-        return BeGoneSettingsHelper.isBlacklist(this.settings) && !RegExp(BeGoneSettingsHelper.blackliststring(this.settings)).test(document.URL);
+        return BeGoneSettingsHelper.isBlacklist(this.settings) && !BeGoneSettingsHelper.blacklistRegexp(this.settings).test(document.URL);
+    }
+
+    private probeDocument(doc: Document | Element) {
+        return this.probeDocumentContent(superPowerfulTextContentOf(doc))
     }
 
-    private probeDocument(bodyTextContent: string = document.body.textContent ? document.body.textContent : ""):
+    /**
+     * Prüft, ob der Text, generell infrage kommt für Veränderungen.
+     */
+    private probeDocumentContent(bodyTextContent: string):
         {
             probeBinnenI: boolean,
             probeRedundancy: boolean,
             probePartizip: boolean,
             probeGefluechtete: boolean,
             probeArtikelUndKontraktionen: boolean;
-
         } {
-        let probeBinnenI = false;
-        let probeRedundancy = false;
-        let probePartizip = false;
-        let probeGefluechtete = false;
-        let probeArtikelUndKontraktionen = false;
+        const result = {
+            probeBinnenI: false,
+            probeRedundancy: false,
+            probePartizip: false,
+            probeGefluechtete: false,
+            probeArtikelUndKontraktionen: false
+        };
 
-        if (!this.settings.skip_topic || this.settings.skip_topic && this.mtype || this.settings.skip_topic && !/Binnen-I|Geflüchtete/.test(bodyTextContent)) {
-            probeBinnenI = /[a-zäöüß]{2}((\/-?|_|\*|:|\.|\u00b7| und -)?In|(\/-?|_|\*|:|\.|\u00b7| und -)in(n[\*|\.]en)?|(\/-?|_|\*|:|\.|\u00b7)ze||(\/-?|_|\*|:|\.|\u00b7)a|(\/-?|_|\*|:|\.|\u00b7)nja|INNen|\([Ii]n+(en\)|\)en)?|\/inne?)(?!(\w{1,2}\b)|[A-Z]|[cf]o|t|act|clu|dex|di|line|ner|put|sert|stall|stan|stru|val|vent|v?it|voice)|[A-ZÄÖÜß]{3}(\/-?|_|\*|:|\.)IN\b|(der|die|dessen|ein|sie|ihr|sein|zu[rm]|jede|frau|man|eR\b|em?[\/\*.&_\(])/.test(bodyTextContent);
-            probeArtikelUndKontraktionen = /[a-zA-ZäöüßÄÖÜ][\/\*.&_\(]-?[a-zA-ZäöüßÄÖÜ]/.test(bodyTextContent) || /der|die|dessen|ein|sie|ihr|sein|zu[rm]|jede|frau|man|eR\b|em?[\/\*.&_\(]-?e?r\b|em?\(e?r\)\b/.test(bodyTextContent);
+        if (!this.settings.skip_topic || (this.settings.skip_topic && (this.mtype == "ondemand" || !/Binnen-I|Geflüchtete/.test(bodyTextContent)))) {
+            result.probeBinnenI = /[a-zäöüß]{2}((\/-?|_|\*|:|\.|\u00b7| und -)?In|(\/-?|_|\*|:|\.|\u00b7| und -)in(n[*|.:]en)?|(\/-?|_|\*|:|\.|\u00b7)ze|(\/-?|_|\*|:|\.|\u00b7)[ar]|(\/-?|_|\*|:|\.|\u00b7)nja|INNen|\([Ii]n+(en\)|\)en)?|\/inne?)/.test(bodyTextContent)
```

Comment:
Hier war ein Fehler, irgendwo in dem Ausdruck von "probeBinnenI" war "||" versteckt. 
Das hat dann natürlich alles gematcht.

Habe ich korrigiert, dann funktionierte das probeGefluechtete  nicht mehr, daher workaround paar Zeilen weiter.